### PR TITLE
Localize all timestamps

### PR DESCRIPTION
**Pull request by yardenshoham**
Following
* #21410 

We are now able to localize all timestamps. Some examples:

`short-date` format, French, user profile page:
image

`date-time` format, Portuguese, mirror repository settings page:
image

Signed-off-by: Yarden Shoham <hrsi88@gmail.com>

### CODE REVIEW

**Comment by silverwind** [comment]
We may be able to do this on backend with https://github.com/go-gitea/gitea/pull/21443. I think it's certainly better to do it there to avoid content flashes. If we want to keep date formatting in frontend, there is the option of https://github.com/go-gitea/gitea/pull/21429 which I devised.

**Comment by yardenshoham** [comment]
I read both PRs you linked. About the following comment:

        `requestAnimationFrame` is the proper way to mutate DOM without flicker. It can and should be avoided. It's just sloppiness on GitHub's side that they still have these flickers.

        I will also move all stuff in [formatting.js](https://github.com/yardenshoham/gitea/blob/main/web_src/js/features/formatting.js) into this script as well. These are all critical replacements.

_Originally posted by @silverwind in https://github.com/go-gitea/gitea/issues/21429#issuecomment-1277119954_

If you are still doing that then it's worth merging this as it's simply building on
* #21410

Any solution of a different implementation would still have to include this JS so I suggest merging this for now

#21410 did this for the activity page, this just does it for all other timestamps, easy win

**Comment by silverwind** [comment]
If we decide for the JS solution, we can certainly go ahead with this already.

**Comment by yardenshoham** [comment]
* I'm hoping to merge this, I believe having better datetime localization is a big plus even if it comes with a lightweight flicker. The work on #21429 can keep going, this doesn't hinder it at all

File: web_src/js/features/formatting.js [hunk 1]
```diff
@@ -13,9 +16,34 @@ export function initFormattingReplacements() {
...
+    case 'short-date':
+      formatter = shortDateFormatter;
+      break;
+    case 'date-time':
+      formatter = dateTimeFormatter;
+      break;
+    default:
+      throw new Error('Unknown format');
+  }
+  for (const timeElement of document.querySelectorAll(`time[data-format="${format}"]`)) {
```
Comment (wxiaoguang) [inline web_src/js/features/formatting.js]: Use `document.querySelectorAll("time[data-format]")` to get all elements in one shot.

**Review by wxiaoguang — APPROVED** [review]
I can vote LGTM

File: web_src/js/features/formatting.js (hunk 1 above)
Comment (yardenshoham) [reply web_src/js/features/formatting.js]: Right, I don't want to get all in one shot

Comment (wxiaoguang) [reply web_src/js/features/formatting.js]: Why not? If you take a test, I think `document.querySelectorAll("time[data-format]") ` could be faster.

Comment (yardenshoham) [reply web_src/js/features/formatting.js]: Ok, done

File: web_src/js/features/formatting.js [hunk 2]
```diff
@@ -13,9 +16,28 @@ export function initFormattingReplacements() {
     }
   }
 
-  // for each <time></time> tag, if it has the data-format="date" attribute, format
-  // the text according to the user's chosen locale
-  for (const timeElement of document.querySelectorAll('time[data-format="date"]')) {
-    timeElement.textContent = dateFormatter.format(new Date(timeElement.dateTime));
+  // for each <time></time> tag, if it has the data-forma attribute, format
```
Comment (Gusted) [inline web_src/js/features/formatting.js]:
```suggestion
  // for each <time></time> tag, if it has the data-format attribute, format
```

File: web_src/js/features/formatting.js [hunk 3]
```diff
@@ -13,9 +16,28 @@ export function initFormattingReplacements() {
...
+  // the text according to the user's chosen locale and formatter
```
Comment (Gusted) [inline web_src/js/features/formatting.js]:
```suggestion
  // the text according to the user's chosen locale and formatter.
```

File: web_src/js/features/formatting.js [hunk 4]
```diff
@@ -13,9 +16,28 @@ export function initFormattingReplacements() {
...
+  formatAllTimeElements();
+}
+
+function formatAllTimeElements() {
+  const allTimeElements = document.querySelectorAll('time[data-format]');
+  for (const timeElement of allTimeElements) {
```
Comment (Gusted) [inline web_src/js/features/formatting.js]: `all` is a repeating if you have Element**s** already in the variable name.
```suggestion
  const timeElements = document.querySelectorAll('time[data-format]');
  for (const timeElement of timeElements) {
```

**Review by Gusted — APPROVED** [review]
Alright LGTM. Given the other PR to make the JS execution faster, the concern of FOUC shouldn't be *that* big, if the other PR is merged before 1.18

File: templates/admin/auth/list.tmpl [hunk 5]
```diff
@@ -29,8 +29,8 @@
 							<td><a href="{{AppSubUrl}}/admin/auths/{{.ID}}">{{.Name}}</a></td>
 							<td>{{.TypeName}}</td>
 							<td>{{if .IsActive}}{{svg "octicon-check"}}{{else}}{{svg "octicon-x"}}{{end}}</td>
-							<td><span class="tooltip" data-content="{{.UpdatedUnix.FormatShort}}">{{.UpdatedUnix.FormatShort}}</span></td>
-							<td><span class="tooltip" data-content="{{.CreatedUnix.FormatLong}}">{{.CreatedUnix.FormatShort}}</span></td>
+							<td><span class="tooltip" data-content="{{.UpdatedUnix.FormatShort}}"><time data-format="short-date" datetime="{{.UpdatedUnix.FormatLong}}">{{.UpdatedUnix.FormatShort}}</time></span></td>
+							<td><span class="tooltip" data-content="{{.CreatedUnix.FormatLong}}"><time data-format="short-date" datetime="{{.CreatedUnix.FormatLong}}">{{.CreatedUnix.FormatShort}}</time></span></td>
```
Comment (delvh) [inline templates/admin/auth/list.tmpl:33]: Hmm, should we perhaps extract this `<span>` into its own template?
I think that should be possible in this case…
However, we can probably defer that for a little and merge this PR first…

Comment (yardenshoham) [reply templates/admin/auth/list.tmpl:33]: Nice thinking but I think it's small enough so that its own template will be overkill

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**